### DISCUSSION TITLE
ERROR: std::bad_alloc for whole planet import

### DISCUSSION THREAD

**Discussion by MALKARAJ** [osm-search/Nominatim#3296] [Q&A]
**nominatim importer hardware specification**

nominatim version v4.2.3
RAM 64GB
HARDDISK SIZE 1.5 TB
CPU CORE 32

**postgres hardware specification**

postgres 15 

master, slave, sec slave each  with these specification
RAM 128GB
DISK 3.1 TB
CPU CORES 32
postgis-3.3.3

```
log

2024-01-08 10:17:29  ERROR: std::bad_alloc
Traceback (most recent call last):
  File "/usr/local/bin/nominatim", line 14, in 
    exit(cli.nominatim(module_dir='/usr/local/lib/nominatim/module',
  File "/usr/local/lib/nominatim/lib-python/nominatim/cli.py", line 264, in nominatim
    return parser.run(**kwargs)
  File "/usr/local/lib/nominatim/lib-python/nominatim/cli.py", line 126, in run
    return args.command.run(args)
  File "/usr/local/lib/nominatim/lib-python/nominatim/clicmd/setup.py", line 94, in run
    database_import.import_osm_data(files,
  File "/usr/local/lib/nominatim/lib-python/nominatim/tools/database_import.py", line 113, in import_osm_data
    run_osm2pgsql(options)
  File "/usr/local/lib/nominatim/lib-python/nominatim/tools/exec_utils.py", line 159, in run_osm2pgsql
    subprocess.run(cmd, cwd=options.get('cwd', '.'),
  File "/usr/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/local/lib/nominatim/osm2pgsql', '--hstore', '--latlon', '--slim', '--log-progress', 'true', '--number-processes', '1', '--cache', '94485', '--style', '/usr/local/etc/nominatim/import-full.style', '--output', 'gazetteer', '--create', '--with-forward-dependencies', 'false', '/nominatim/data.osm.pbf']' returned non-zero exit status 1.``
```

**is the hardware specs is sufficient to run planet import process ?**

**Comment by MALKARAJ** [comment]
actually it is an issue

**Comment by mtmail** [comment]
128GB is recommended, see https://nominatim.org/release-docs/latest/admin/Installation/#hardware but that's usually for having Nominatim and the database on the same host so I'm not sure what the exact requirement is if both are installed on separate hosts. It makes sense to watch the RAM usage during installation with monitoring software.

You should use the flatnode file feature for the import. I'm not sure from the output you posted if you did that. See https://nominatim.org/release-docs/latest/admin/Import/#flatnode-files

**Reply by MALKARAJ** [reply]
Hi @mtmail  how to enable flatnode-file in docker version

**Reply by mtmail** [reply]
It's explained in their documentation.
